Source/AssetRipper.GUI.Electron/Pages/Assets/FontTabContent.cshtml

```diff
@@ -0,0 +1,94 @@
+﻿@model ValueTuple<byte[], string>
+
+@if (!string.IsNullOrEmpty(Model.Item2))
+{
+    <p>This file type: @Model.Item2</p>
+
+    <script>
+        class FontPreviewer extends HTMLElement {
+            #shadowRoot;
+            #uniqName;
+            #data;
+
+            constructor() {
+                super();
+                this.#shadowRoot = this.attachShadow({ mode: "open" });
+                this.#uniqName = this.randName(10);
+            }
+
+            static get observedAttributes() {
+                return ["data"];
+            }            
+
+            connectedCallback() {
+                this.#data = Uint8Array.from(atob(this.getAttribute("data")), c => c.charCodeAt(0));
+                this.attachFont().then(() => this.attachStyle());
+                this.render();
+
+                console.log(this.#data);
+            }
+
+            attributeChangedCallback(name, oldValue, newValue) {
+                if (name === "data") {
+                    this.#data = Uint8Array.from(atob(newValue), c => c.charCodeAt(0));
+                    this.attachFont().then(() => this.attachStyle());
+                    this.render();
+
+                    console.log(this.#data);
+                }
+            }
+
+            async attachFont() {
+                const font = new FontFace(this.#uniqName, this.#data);
+                await font.load();
+                document.fonts.add(font);
+            }
+
+            async attachStyle(useCSSStyleSheet = true) {
+                const styleContent = `:host { font-family: "${this.#uniqName}"; }`;
+
+                if (!!window.StyleSheet && useCSSStyleSheet) {
+                    const sheet = new CSSStyleSheet();
+                    await sheet.replace(styleContent);
+                    this.#shadowRoot.adoptedStyleSheets = [sheet];
+                } else {
+                    const style = document.createElement("style");
+                    style.textContent = styleContent;
+                    this.#shadowRoot.appendChild(style);
+                }
+            }
+
+            render() {
+                let container = this.#shadowRoot.querySelector(".font-container");
+                    
+                if (!container) {
+                    container = document.createElement("div");
+                    container.className = "font-container";
+                    this.#shadowRoot.appendChild(container);
+                }
+
+                container.innerHTML = "<slot></slot>";
+            }
+
+            randName(length) {
+                let str = '';
+                for (let i = 0; i < length; i++) {
+                    let rand = Math.random() * (62 - 10) + 10 << 0;
+                    str += String.fromCharCode(rand += rand > 9 ? (rand < 36 ? 55 : 61) : 48);
+                }
+                return str;
+            }
+        }
+        customElements.define("font-view", FontPreviewer);
+    </script>
```

Comment:
I don't know enough HTML and JavaScript to know if this is a good way to do things. To me, it seems very verbose.

From a brief look online, it seems like we could do this:

```html
<style>
@font-face {
 font-family: loadedfont;
 src=url (data:font/@FileExtension;base64,@Convert.ToBase64String(Model.Item1, Base64FormattingOptions.None));
}
</style>
<p style="font-size: 12px, font-family: loadedfont"><span class="label">12: </span>Preview: 1234567890</p>
```
Reference:
https://www.wikihow.com/Upload-Your-Own-Fonts-to-HTML-Using-CSS#:~:text=Add%20your%20font%20file.,and%20EOT%20font%2Dfile%20formats.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types

Reply:
While the CSS method is simpler, the Web Component approach offers several benefits that can be crucial as our project evolves:

1. Encapsulation with shadow DOM to avoid style conflicts.
2. Dynamic Updates allowing for more interactive experiences.
3. Unique Font Naming to prevent font conflicts.
4. Extendability for future improvements and functionalities.